db/migrate/20170327100309_lock_plugin_roles.rb

```diff
@@ -0,0 +1,15 @@
+class LockPluginRoles < ActiveRecord::Migration
+  def up
+    Foreman::Plugin.all.each do |plugin|
+      plugin.default_roles.each do |role_name, expected_perms|
+        role = Role.find_by :name => role_name
+        next unless role
```

Comment:
how can this happen? if we skip the locking here, the migration will be marked as done and won't run again so if this is caused by some "db not ready" issue, this would leave the role unlocked